src/main/java/com/spotify/reaper/service/SegmentRunner.java

```diff
@@ -223,6 +225,7 @@ private void abort(RepairSegment segment, JmxProxy jmxConnection) {
   @Override
   public void handle(int repairNumber, ActiveRepairService.Status status, String message) {
     synchronized (condition) {
+      Thread.currentThread().setName(clusterName);
```

Comment:
Since you set the thread name in JmxProxy.handleNotification, you shouldn't have to set it in here again. So we don't need to make any changes in SegmentRunner.